### DISCUSSION TITLE
Read/write memory from cli invocation

### DISCUSSION THREAD

**Discussion by honi** [mortbopet/Ripes#270] [Q&A]
I don't think this is currently possible, but maybe I misread the docs or there's some undocumented way.

I want to run an assembly program through ripes in cli mode. I saw that one can set initial register values, but is it possible to set some memory addresses as well?

Likewise after execution, is there a way to read some memory addresses?

Use case is that we're going to use ripes in a course and we'd like to prepare verification scripts so that the students can use ripes in gui mode to develop their program in assembly, and then run the script to validate their program is correct. The script would execute ripes in cli mode and then verify the program worked correctly. Some assignments require modifying memory in some way.

**Comment by mortbopet** [comment]
No, and no :) ... but, I don't think it would be too difficult to add! 

Register initialization for CLI is defined here: https://github.com/mortbopet/Ripes/blob/565026f3b39c2151f3ba413e8192a7866b3e244a/src/cli/clioptions.cpp#L137-L138
Which is used during processor selection: https://github.com/mortbopet/Ripes/blob/565026f3b39c2151f3ba413e8192a7866b3e244a/src/cli/clirunner.cpp#L38-L39
... stored here:
https://github.com/mortbopet/Ripes/blob/master/src/processorhandler.cpp#L285
and those values are then written every time the processor resets:
https://github.com/mortbopet/Ripes/blob/master/src/processorhandler.cpp#L273

You could do something similar, using memory writing
(https://github.com/mortbopet/Ripes/blob/master/src/processorhandler.cpp#L142)
 and reading
(https://github.com/mortbopet/Ripes/blob/master/src/processorhandler.cpp#L383).

looking forward to the PR!
